### DISCUSSION TITLE
Adding background components doesn't affect signal-to-noise ratio

### DISCUSSION THREAD

**Discussion by LeoJuh** [flatironinstitute/CaImAn#1294] [Q&A]
In four different iterations of CaImAn analysis of the same t-series, I varied the number of background components from one to four. However, when I thereafter calculated the SNR from the iterations, they all had roughly the same SNR. This doesn't make sense to me since it was written in the documentation that a too high number of background components will absorb signal and decrease SNR. Anyone have any idea why my result looks like this?

![caiman_mean peak signal-to-noise](https://github.com/flatironinstitute/CaImAn/assets/89864831/1c50ce2c-fd6d-488f-ba12-71742fefb3b3)

**Comment by EricThomson** [comment]
This is a great question. Four is not very high.  I tend to think this will happen when you set it really high, like 50. I wonder what this plot will look like if you go from 1-50 (in like 5-10 step increments).  I'd be really curious what the background spatial components look like in those cases.  To my knowledge this hasn't been systematically explored (though I'm sure Eftychios and Andrea looked at it in detail, it's just not published to my knowledge: @j-friedrich may have looked at this too). 

I personally have not played with the background number parameter (`nb`) systematically.  There are two bad things that in theory can happen: it can steal signal from neurons that still end up in the final set of components but with lower SNR. It could also steal neural components altogether, so they no longer end up in your components at all (i.e., neurons become part of the background model).

It's tricky to evaluate plots like yours because you'd ideally want to make sure you were looking at the same neural component across comparisons, and as `nb` increases my *guess* is you will start to lose components. But this would also be very useful to know.  

You are looking at gross statistical changes, and it might be natural to think that the loss of signal should just show up as decreased SNR, but it may be more subtle than that. For instance, if you pick one component and plot it for different `gnb` values how does the trace actually look? Not just at the high magnitude parts (which determines SNR), but for the entire trace? These would be very interesting things to see across a range of `nb` values. 

Edit: I'm speculating about things that might happen. As I said I haven't explored this, so would be very curious to see. 🚀

**Reply by LeoJuh** [reply]
Thank you for the answer!

**Comment by kushalkolar** [comment]
Looking at the reconstructed background movie, `b * f`, is a better way to see if your real signal is being absorbed into the background.

(You should really be looking at the reconstructed movie and residuals as well for quality inspection of results)

**Reply by LeoJuh** [reply]
Thx for the reply. What do you mean with residuals?

**Reply by EricThomson** [reply]
`Raw Movie = Neural Model + Background Model + Residual`

The revised demo notebooks go into this in some detail and show how to plot all of these things -- including the residual (the stuff that Caiman's model of neural + background activity doesn't explain), and why they are important.

In practice, these are interesting academic questions about `nb` and CNMF, but not super practical. That is, if you are tempted to crank up `nb` past 3 then you should probably be using CNMFE.
